src/web/app/pages-instructor/instructor-student-list-page/instructor-student-list-page.component.html

```diff
@@ -62,20 +62,25 @@ <h4 class="card-text inline"> teams</h4>
                       (removeStudentFromCourseEvent)="removeStudentFromCourse(courseTab, $event)">
                   </tm-student-list>
                 </ng-container>
+                <ng-template #noStudentsTemplate>
+                  <ng-container *ngIf="courseTab.isAbleToViewStudents, else noViewStudentsPermissionTemplate">
+                    <h5>There are no students in this course.</h5>
+                    <a id="btn-enroll" class="btn btn-success"
+                       tmRouterLink="/web/instructor/courses/enroll"
+                       [queryParams]="{courseid: courseTab.course.courseId}">
+                      Enroll Students
+                    </a>
+                  </ng-container>
+                </ng-template>
+                <ng-template #noViewStudentsPermissionTemplate>
+                  <div class="alert alert-danger" role="alert">
+                    <h5>You do not have permission to view the details of the students in this course.</h5>
+                  </div>
+                </ng-template>
               </div>
             </div>
           </tm-loading-retry>
         </div>
-
-        <ng-template #noStudentsTemplate>
```

Comment:
I think while the elements were not all within the same level, the code before my changes was still okay since it was only one (ng) if-else statement, whether there are students or not, so it can still be kind of clear.

My solution essentially adds another if-else statement, whether permission is granted or not. I felt that since there are now two if-else statements, it would be clearer if I put all those elements within the same level.

Reply:
You can still put the if-else under this block and it's not any less clearer; the `noStudentsTemplate` is still only called by one place only. In fact, putting if-else block deeper contributes to the arrowhead anti-pattern.

But I am willing to close one eye here.

Reply:
Fair enough. I can move them down then.